distribution_strategy/multi_worker_mirrored_strategy/examples/custom_training_mnist.py

```diff
@@ -0,0 +1,170 @@
+# ==============================================================================
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+# This code serves as an example of using Tensorflow 2.0 to build and train a CNN model on the 
+# Fashion MNIST dataset using the tf.distribute.MultiWorkerMirroredStrategy described here 
+# https://www.tensorflow.org/api_docs/python/tf/distribute/experimental/MultiWorkerMirroredStrategy 
+# using a custom training loop. This code is very similar to the example provided here
+# https://www.tensorflow.org/tutorials/distribute/custom_training
+# Assumptions: 
+#   1) The code assumes that the cluster configuration needed for the TF distribute strategy is available through the 
+#   TF_CONFIG environment variable. See the link provided above for details
+#   2) The model is checkpointed and saved in /pvcmnt by the chief worker process.
+
+import tensorflow as tf
+import numpy as np
+import os
+
+# Used to run example using CPU only. Untested on GPU
+os.environ["CUDA_VISIBLE_DEVICES"] = "-1"
+MAIN_MODEL_PATH = '/pvcmnt'
+
+EPOCHS = 10
+BATCH_SIZE_PER_REPLICA = 64
```

Comment:
Let's define `GLOBAL_BATCH_SIZE` as a constant, instead of computing it from a constant BATCH_SIZE_PER_REPLICA + strategy. num_replicas_in_sync.

One of the key idea of distribution strategy is to allow users to define a fixed global batch size, and provide a mathematically identical training process regardless of how many replicas they use. For example if user wants to add more machines, they can just change TF_CONFIG, and there is no need to modify the code.